### ISSUE TITLE
jsx-curly-spacing should support "consistent" option

### ISSUE THREAD

**Issue by kaiyoma** [yannickcr/eslint-plugin-react#1493]
Similar to how other eslint rules provide a "consistent" option for square brackets, curly braces, and parens, there should be a similar option for the curly braces in JSX. Basically, the absence or presence of a newline after the opening curly brace should be matched at the closing curly brace.

**Comment by ljharb** [comment]
Can you provide some code samples that it should warn on, and what it would autofix to?

**Comment by kaiyoma** [comment]
No newlines at either end is good:

```
  <div>
    {data.message ? <span>You have an unread message</span> : <Placeholder/>}
  </div>
```

Newlines at both ends is also good:

```
  <div>
    {
      data.message ?
        <span>You have an unread message</span> :
        <Placeholder/>
    }
  </div>
```

But mixing and matching is bad:

```
  <div>
    {
      data.message ?
        <span>You have an unread message</span> :
        <Placeholder/>}
  </div>
```

```
  <div>
    {data.message ?
        <span>You have an unread message</span> :
        <Placeholder/>
    }
  </div>
```

Related eslint rules that support `"consistent"`:

- https://eslint.org/docs/rules/array-bracket-newline
- https://eslint.org/docs/rules/function-paren-newline
- https://eslint.org/docs/rules/object-curly-newline

**Comment by ljharb** [comment]
Sounds like a great addition! The airbnb config would use this option.

**Comment by kamronbatman** [comment]
Definitely need this.  There is also no rule to enforce a newline if the JSX curly brace block is multiline.

**Comment by ljharb** [comment]
The third "good" is the one I'd want.

**Comment by kamronbatman** [comment]
I agree with you @ljharb.  Currently I cannot enforce that with eslint-plugin-react.  I can try and find some time to do a dirty addition, but I am not familiar with writing eslint rules. It might go faster with someone else who is more familiar.

**Comment by viktor-equinix** [comment]
Is anybody found the way how to configure eslint for 3rd example above?

**Comment by kamronbatman** [comment]
This rule effectively doesn't exit. :(

**Comment by JBallin** [comment]
Should also enforce consistent spacing around content:

**Comment by ljharb** [comment]
A PR to implement this is most welcome.
